CommonTools/UtilAlgos/interface/EventSelectorAdapter.h

```diff
@@ -17,6 +17,38 @@
 #include "FWCore/Framework/interface/Event.h"
 #include "FWCore/Framework/interface/EventSetup.h"
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescriptionFiller.h"
+
+template <typename T>
+struct SelectorFillDescriptions {
+};
+
+namespace impl {
+  template <typename SFD>
+  struct FillDescriptions {
+    void operator()(edm::ConfigurationDescriptions& descriptions) {
+      SFD::fillDescriptions(descriptions);
+    }
+  };
+  struct DoDefault {
+    void operator()(edm::ConfigurationDescriptions& descriptions) {
+      // Same default as framework itself as I don't know how I could
+      // avoid defining fillDescriptions() depending whether a "trait"
+      // class template has fillDescriptions() or not.
+      edm::fillDetails::DoFillAsUnknown<void> filler; // AFAICT DoFillAsUnknown wouldn't have to be a template, so passing just void here
+      filler(descriptions);
+    }
+  };
+}
+
+template <typename T>
+void selectorFillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  using Traits = SelectorFillDescriptions<T>;
+  std::conditional_t<edm::fillDetails::has_fillDescriptions_function<Traits>::value,
+                     impl::FillDescriptions<Traits>,
+                     impl::DoDefault> fill_descriptions;
+  fill_descriptions(descriptions);
+}
```

Comment:
If we take this route, all the above should be moved to a header of it's own to be included in all selector headers that define an EDModule.